packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -634,6 +634,17 @@ export function unhideTextInstance(
   textInstance.nodeValue = text;
 }
 
+export function clearContainer(container: Container): void {
+  if (container.nodeType === ELEMENT_NODE) {
+    ((container: any): Element).innerHTML = '';
+  } else if (container.nodeType === DOCUMENT_NODE) {
+    const documentElement = ((container: any): Document).documentElement;
+    if (documentElement != null) {
+      documentElement.innerHTML = '';
```

Comment:
I wonder if we should ever do this? Shouldn't we trigger an invariant if mounting into the `document` instead?

Reply:
React DOM's `Container` type is `Element | Document`. I don't know the historical reasons for supporting the latter, but that change seems much larger scope than this PR.

Reply:
That being said, maybe this change is too aggressive anyway for the Document container case, since it would also clear `<head>` tags. I wish I knew more about why we supported that in the first place.

Reply:
We don't clear the `document` in the legacy case, so we should probably not do it here either.

Reply:
Where do you see that?

https://github.com/facebook/react/blob/4b02b6611141255a8e8b6d8a997e87c865f06626/packages/react-dom/src/client/ReactDOMLegacy.js#L113-L138

Looks to me like it removes all element nodes in the container, regardless of the container type?

Reply:
That isn't the same as what you're doing. You're clearing the HTML, that above code is clearing the body contents of the document. When you do `document.lastChild` it gets the node from the body.

Reply:
Ignore me, I noticed you're clearing `documentElement`. That should be fine here.

Reply:
Your edited comment makes sense. The previous one
> We don't clear the document in the legacy case, so we should probably not do it here either.

Made me think that you were suggesting I should *nothing* in the `DOCUMENT_NODE` case.

Anyway you're commenting on an outdated commit. I had already changed it to clear only the body.

Reply:
Yeah I noticed. The mobile UI for Github sucks though. Sorry for the churn.

Reply:
It does!

No worries.